app/javascript/packs/new_service.js

```diff
@@ -0,0 +1,9 @@
+import {NewServiceFormWrapper} from 'NewService'
+import {safeFromJsonString} from 'utilities/json-utils'
+
+document.addEventListener('DOMContentLoaded', () => {
+  const newServiceWrapper = document.getElementById('new_service_wrapper')
+  const newServiceData = safeFromJsonString(newServiceWrapper.dataset.newServiceData) || {}
```

Comment:
Can't you work with the `undefined` value returned in the case it failed from `safeFromJsonString`